### PR TITLE
[prompty] Expose Prompty class as public

### PR DESCRIPTION
**Pull request by lalala123123**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

# All Promptflow Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: examples/prompty/basic/prompty-quickstart.ipynb [hunk 1]
```diff
@@ -79,10 +79,10 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "from promptflow.core import Flow\n",
+    "from promptflow.core import Prompty\n",
```
Comment (wangchao1230) [inline examples/prompty/basic/prompty-quickstart.ipynb:82]: please change sample in separte PR: https://github.com/microsoft/promptflow/pull/2856

File: src/promptflow-core/promptflow/core/__init__.py [hunk 2]
```diff
@@ -7,7 +7,7 @@
 
 # flake8: noqa
 from promptflow._core.tool import ToolProvider, tool
-from promptflow.core._flow import AsyncFlow, Flow
+from promptflow.core._flow import AsyncFlow, AsyncPrompty, Flow, Prompty
```
Comment (wangchao1230) [inline src/promptflow-core/promptflow/core/__init__.py:10]: change the prompty docstring a bit: 
from promptlfow.core import AzureOpenAIModelConfiguration
image

File: src/promptflow-core/promptflow/core/__init__.py [hunk 3]
```diff
@@ -25,6 +25,8 @@
     "tool",
```
Comment (wangchao1230) [inline src/promptflow-core/promptflow/core/__init__.py:25]: write await here
image

**Review by Adrian999999999 — APPROVED** [review]

File: src/promptflow-core/promptflow/core/__init__.py (hunk 3 above)
Comment (lalala123123) [reply src/promptflow-core/promptflow/core/__init__.py:25]: fix it

**Review by Adrian999999999 — APPROVED** [review]

**Review by wangchao1230 — DISMISSED** [review]

File: src/promptflow-core/promptflow/core/__init__.py (hunk 2 above)
Comment (lalala123123) [reply src/promptflow-core/promptflow/core/__init__.py:10]: fix it

File: src/promptflow-core/promptflow/core/__init__.py [hunk 4]
```diff
@@ -25,6 +25,8 @@
...
     "Flow",
     "AsyncFlow",
+    "Prompty",
```
[the remark this answers is not in the document]
Comment (lalala123123) [reply src/promptflow-core/promptflow/core/__init__.py:28]: remove unused param

**Review by elliotzh — APPROVED** [review]

**Review by wangchao1230 — APPROVED** [review]

**Pull request merged**